### ISSUE TITLE
`Push origin` Error: "fatal: could not read Username [...] terminal prompts disabled"

### ISSUE THREAD

**Issue by chriddyp** [desktop/desktop#1514]
Love the new app! I just tried to push to a Repo and I got this error:

![image](https://cloud.githubusercontent.com/assets/1280389/26126923/7a83b33a-3a55-11e7-978f-f99f52c13bae.png)

```
env: illegal option -- u
usage: env [-i] [name=value ...] [utility [argument ...]]
error: unable to read askpass response from '/Users/chriddyp/Downloads/GitHub Desktop.app/Contents/Resources/app/static/ask-pass-trampoline.sh'
fatal: could not read Username for 'https://github.com': terminal prompts disabled
```

Not sure what's going wrong here. I successfully logged in through GitHub's Ouath (with 2 factor).

### Version

**GitHub Desktop version:** 0.5.3

**OS version:** Mac 10.10.5

### Steps to Reproduce

1. Added a local version of https://github.com/plotly/plotly-database-connector/ to GitHub Desktop
2. Made a simple commit on a branch
3. Clicked `Push origin`
4. Got this error.

Repo: https://github.com/plotly/plotly-database-connector/

If you think screenshots or a GIF recording will help demonstrate the issue
better, feel free to add them here.

**Expected behaviour:** Push should have succeeded

**Actual behaviour:** Error message

**Reproduces how often:** Every time I try to push. I haven't tried with any other repos yet.

### Logs

There were no logs:
```
$ ls ~/Library/Application\ Support/GitHub\ Desktop/logs
ls: /Users/chriddyp/Library/Application Support/GitHub Desktop/logs: No such file or directory
```

#### Additional Information

The non-beta version of GitHub Desktop works for me.

Thanks for building such great software!

**Comment by shiftkey** [comment]
@chriddyp thanks for the report. For reference, the `askpass` response is this code, which is just poking at environment variables we pass from the parent process to Git when we invoke it.

https://github.com/desktop/desktop/blob/3dc8738e711131047f9973398a5dac08e2cd206e/app/src/ask-pass/ask-pass.ts

Stepping through the output:

```
env: illegal option -- u
```

Not sure where this is coming from. 🤷‍♂️ 

```
error: unable to read askpass response from '/Users/chriddyp/Downloads/GitHub Desktop.app/Contents/Resources/app/static/ask-pass-trampoline.sh'
```

Looks like we didn't return any response from the ask-pass script, which is reasonable if we don't have any credentials.

```
fatal: could not read Username for 'https://github.com': terminal prompts disabled
```

Yeah, we [turn off the terminal prompt](https://github.com/desktop/desktop/blob/3dc8738e711131047f9973398a5dac08e2cd206e/app/src/lib/git/core.ts#L233) because Git needs to work in a headless mode. So this is fine.

> I successfully logged in through GitHub's Ouath (with 2 factor).

Could you confirm this completed successfully? The account should be there in the menu under `GitHub Desktop` -> `Preferences`.

**Comment by chriddyp** [comment]
> Could you confirm this completed successfully? The account should be there in the menu under GitHub Desktop -> Preferences.

Looks like it's OK:

![image](https://cloud.githubusercontent.com/assets/1280389/26127919/e4bd8cdc-3a58-11e7-8594-b4d3bbe08504.png)

**Comment by chriddyp** [comment]
However I notice that the Avatar for the commit messages and the history don't match my github avatar. Not sure if that's indicative of the accounts not syncing correctly or something. 

![image](https://cloud.githubusercontent.com/assets/1280389/26128024/1f3a0b56-3a59-11e7-986e-d50c4b532339.png)
![image](https://cloud.githubusercontent.com/assets/1280389/26128047/2ff49c36-3a59-11e7-8608-3aa3d469075e.png)

**Comment by sebastian-meier** [comment]
I see the exact same behavior in my github desktop client. 
I have added through drag and drop, as well as folder selection.
I have tried multiple repos, all throw the same error.
Although in comparison to @chriddyp my user icon is fine.

**Comment by sebastian-meier** [comment]
I also cloned a remote repository, did some changes and get the same error message. Please let me know if you need further info on os etc.

**Comment by niik** [comment]
> Not sure where this is coming from. 🤷‍♂️

@shiftkey I'm guessing that would be from [the trampoline](https://github.com/desktop/desktop/blob/ab8556184fbf5d978722e61fee18866b80203764/app/static/darwin/ask-pass-trampoline.sh#L6).

@joshaber Would know best but that flag isn't documented in the [macOS man page for env](https://developer.apple.com/legacy/library/documentation/Darwin/Reference/ManPages/man1/env.1.html) though it is in [the unix one](https://linux.die.net/man/1/env)

**Comment by joshaber** [comment]
I think that could actually go away now that we're [explicitly unsetting the credential helper](https://github.com/desktop/desktop/blob/ab8556184fbf5d978722e61fee18866b80203764/app/src/lib/git/core.ts#L222).

**Comment by niik** [comment]
Agreed, moving to M5

**Comment by shiftkey** [comment]
Yep, I think this `-u` flag is the issue. On my machine `man env` does indicate it supports `-u`.

@chriddyp  @sebastian-meier could you confirm by opening a terminal and running this command:

```
env -u SOME_VALUE_THAT_DOESNT_EXIST
```

I believe the command should fail on your machines, but on mine it succeeds and prints out the environment list.

I'll open a PR to clean this up now, and hopefully that's all the work we need to do here to close this out.

**Comment by chriddyp** [comment]
> @chriddyp @sebastian-meier could you confirm by opening a terminal and running this command:

You got it @shiftkey :
```
♡♡ env -u SOME_VALUE_THAT_DOESNT_EXIST
env: illegal option -- u
usage: env [-i] [name=value ...] [utility [argument ...]]
```

**Comment by shiftkey** [comment]
@chriddyp cool, that makes me feel more confident we'll close this out with #1592

**Comment by sebastian-meier** [comment]
@shiftkey same output on my machine
`env: illegal option -- u
usage: env [-i] [name=value ...] [utility [argument ...]]`

**Comment by shiftkey** [comment]
I think we've got enough here to be confident #1592 is the fix. 

@chriddyp @sebastian-meier this will be available in the next update (`0.5.5`). Please comment here if you still see an issue.
